src/renate/benchmark/datasets/nlp_datasets.py

```diff
@@ -1,78 +1,120 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-import os
-from pathlib import Path
-from typing import Optional, Union
+import logging
+from typing import Optional
 
-import torchtext
-from torchtext.data import to_map_style_dataset
+import datasets
+import torch
+import transformers
 
 from renate import defaults
 from renate.data.data_module import RenateDataModule
-from renate.utils.file import download_folder_from_s3
 
 
-class TorchTextDataModule(RenateDataModule):
-    """Dataset with data from torchtext.
+class _InputTargetWrapper(torch.utils.data.Dataset):
+    """Make a huggingface dataset comply with the `(input, target)` format."""
 
-    Source: https://pytorch.org/text/stable/datasets.html
+    def __init__(self, dataset, target_column: str = "label"):
+        self._dataset = dataset
+        self._target_column = target_column
+
+    def __len__(self):
+        return len(self._dataset)
+
+    def __getitem__(self, idx):
+        item = self._dataset[idx]
+        target = item.pop(self._target_column)
+        return item, target
+
+
+class HuggingfaceTextDataModule(RenateDataModule):
+    """Data module wrapping Huggingface text datasets.
+
+    This is convenience wrapper to expose a hugginface dataset as a `RenateDataModule`. Datasets
+    will be pre-tokenized and will return `input, target = dataset[i]`, where `input` is a
+    dictionary with fields `["input_ids", "attention_mask"]`, and `target` is a tensor.
+
+    We expect the dataset to have a "train" and a "test" split. An additional "validation" split
+    will be used if present. Otherwise, a validation set may be split off of the training data
+    using the `val_size` argument.
 
     Args:
         data_path: the path to the folder containing the dataset files.
-        src_bucket: the name of the s3 bucket. If not provided, downloads the data from original
-            source.
-        src_object_name: the folder path in the s3 bucket.
-        dataset_name: Name of the torchvision dataset.
+        dataset_name: Name of the dataset, see https://huggingface.co/datasets.
+        input_column: Name of the column containing the input text.
+        target_column: Name of the column containing the target (e.g., class label).
+        tokenizer: Tokenizer to apply to the dataset.
+        tokenize_kwargs: Keyword arguments to be passed to the tokenizer.
```

Comment:
It would be good to provide a pointer to the list of the tokenizers and their kwargs.